File: Training/OOPFundamentals/Java/Exercises/after/04 Multiple Constructors/src/Triangle.java
public class Triangle {

    // Length of the three sides of the triangle
    private float a;
    private float b;
    private float c;

    public Triangle(float a, float b, float c) {
        this.a = a;
        this.b = b;
        this.c = c;
    }

    public Triangle(float sideLength) {
        this(sideLength, sideLength, sideLength);
    }

    public float calculatePerimeter() {
        return this.a + this.b + this.c;
    }
}

File: Training/OOPFundamentals/C#/Exercises/after/MultipleConstructors/Triangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MultipleConstructors
{
    //solution
    class Triangle
    {
        private float _a;
        private float _b;
        private float _c;

        public Triangle(float sideLength)
        {
            _a = sideLength;
            _b = sideLength;
            _c = sideLength;
        }

        public Triangle(float a, float b, float c)
        {
            _a = a;
            _b = b;
            _c = c;
        }

        public float CalculatePerimeter()
        {
            return _a + _b + _c;
        }
    }
    //end solution
}
